lib/model/channel.dart

```diff
@@ -572,6 +641,65 @@ class ChannelStoreImpl extends HasUserStore with ChannelStore {
       forStream[event.topicName] = visibilityPolicy;
     }
   }
+
+  /// Handle a [MessageEvent], returning whether listeners should be notified.
+  bool handleMessageEvent(MessageEvent event) {
```

Comment:
I think there's one more event we want to handle. On `ChannelUpdateEvent` when `event.historyPublicToSubscribers` is present, we should forget the topics we've been tracking for that channel, so that they can be refetched on demand. That's because the fetched data only includes topics the user can know about, and [shared/protected-history setting](https://zulip.com/help/channel-permissions#private-channels) affects which topics those are.